Tools/ApplicationDebugger/jacobi/README.md

```diff
@@ -0,0 +1,340 @@
+
+## Building and Running the `jacobi` Program
+
+> Note: if you have not already done so, set up your CLI 
+> environment by sourcing  the setvars script located in 
+> the root of your oneAPI installation. 
+>
+> Linux Sudo: . /opt/intel/oneapi/setvars.sh  
+> Linux User: . ~/intel/oneapi/setvars.sh  
+> Windows: C:\Program Files(x86)\Intel\oneAPI\setvars.bat
+
+### Setup
+
+Preliminary setup steps are needed for the debugger to function.
+Please see the setup instructions in the Get Started Guide based on
+your OS:
+[Linux](https://software.intel.com/en-us/get-started-with-debugging-dpcpp-linux),
+[Windows](https://software.intel.com/en-us/get-started-with-debugging-dpcpp-windows).
+
+
+### Include Files
+
+The include folder is located at
+`%ONEAPI_ROOT%\dev-utilities\latest\include` on your development
+system.
+
+
+### Running Samples In DevCloud
+
+If running a sample in the Intel DevCloud, remember that you must
+specify the compute node (CPU, GPU, FPGA) and whether to run in
+batch or interactive mode.  For the jacobi sample, a node
+with GPU and an interactive shell is recommended.
+
+```
+$ qsub -I -l nodes=1:gpu:ppn=2
+```
+
+For more information, see the Intel® oneAPI
+Base Toolkit Get Started Guide
+(https://devcloud.intel.com/oneapi/get-started/base-toolkit/).
+
+### On a Linux* System
+
+Perform the following steps:
+
+1.  Build the program using the following `cmake` commands.
+    ```
+    $ cd jacobi
+    $ mkdir build
+    $ cd build
+    $ cmake ..
+    $ make
+    ```
+    > Note: The cmake configuration enforces the `Debug` build type.
+
+2.  Run the bugged program:
+    ```
+    $ ./jacobi-bugged <device>
+    ```
+    > Note: `<device>` is the type of the device type to offload the kernel.
+    > Use `cpu`, `gpu`, or `accelerator` to select the CPU, GPU, or the
+    > FPGA emulator device, respectively.  E.g.:
+    ```
+    $ ./jacobi-bugged cpu
+    ```
+
+3.  Start a debugging session:
+    ```
+    $ gdb-oneapi --args jacobi-bugged <device>
+    ```
+
+4.  Clean the program using:
+    ```
+    $ make clean
+    ```
+
+By default, CMake configures the build for Just-in-Time (JIT)
+compilation of the kernel.  However, it also offers an option for
+*Ahead-of-Time* (AoT) compilation.  To compile the kernel
+ahead-of-time for a specific device, set the `DPCPP_COMPILE_TARGET`
+option to the desired device during configuration.  For CPU, use the
+`cpu` value; for FPGA-emu, use the `fpga-emu` value.  Other values are
+assumed to be for GPU and are passed directly to the GPU AoT
+compiler.
+
+> *Hint:* Run `ocloc compile --help` to see available GPU device options.
+
+For example, to do AoT compilation for a `kbl` GPU device:
+
+```
+$ cmake .. -DDPCPP_COMPILE_TARGET=kbl
+```
+
+or for the Gen12 family:
+
+```
+$ cmake .. -DDPCPP_COMPILE_TARGET=gen12LP
+```
+
+> *Note:* AoT compilation is particularly helpful in larger
+> applications where compiling with debug information takes
+> considerably longer time.
```

Comment:
If we remove the AoT options from the CMake file, we should remove this text, as well.
Is there a compilation time issue with this Jacobi sample?

Reply:
Done, no Jacobi compiles fast enough.